### DISCUSSION TITLE
I found an error message "token has invalid claims: token is expired"

### DISCUSSION THREAD

**Discussion by santichoks** [golang-jwt/jwt#377] [Q&A] [answered]
If I set **ExpiresAt** equal to 0 when **SignedString** and after that I **ParseWithClaims** After that it has an error that "_token has invalid claims: token is expired_" It seems like **jwt.ErrTokenInvalidClaims** + ":" + **jwt.ErrTokenExpired**, Why is it not only equal to **jwt.ErrTokenExpired**? And I try to set **ExpiresAt** equal to a day. But it does not have an error.

**Comment by oxisto** [comment]
Could you elaborate what you mean with "equal to 0"? Are you using `time.Now().Add(0)` or are you setting it to `0`? In the first case, the token expires in the instant you create it, thus it is invalid. In the second case it expired on 1.1.1970 since 0 is the UNIX epoch in `time.Time`. If you want your token to NOT expire at all, just do not set the `ExpiresAt`, it will default to `nil`.

With regards to the error, we are using wrapped errors, so you should not compare it directly against `jwt.ErrTokenExpired` but rather you should use the Go `errors.Is` function from the standard library. You can find an example in our examples:

https://github.com/golang-jwt/jwt/blob/6bcdd9d5b6ecb03a80ac123d1a9dc363441cbffe/example_test.go#L181-L183

**Reply by santichoks** [reply]
I set **ExpiredAt** equal to 0 because I am testing about refresh tokens by checking that if the access token expires **errors.Is(err, jwt.ErrTokenExpired)** will create a new token.

**Reply by oxisto** [reply, marked as answer]
Ok got it. as mentioned, in this case you can use `errors.Is(err, jwt.ErrTokenExpired)` to check for an expired token.

**Reply by santichoks** [reply]
An error is still "_token has invalid claims: token is expired_" and can't use errors. **Is(err, jwt.ErrTokenExpired)** to check this error. Have you got a constant variable to check this?

**Reply by oxisto** [reply]
Why can't you use `errors.Is`? It is exactly designed for this use-case. See the example above. 

Can you provide a minimum code example? Otherwise, its difficult to help you.

**Reply by santichoks** [reply]
I apologize for any confusion. I made a mistake. I forgot to add "!" before checking this. I can solve this problem according to what you recommend.

Thank you very much for your answer.

**Reply by oxisto** [reply]
No worries :) Perfect, that it works now!
